diesel/src/pg/expression/array.rs

```diff
@@ -0,0 +1,106 @@
+use std::marker::PhantomData;
+use backend::Backend;
+use expression::{AppearsOnTable, Expression, NonAggregate, SelectableExpression};
+use query_builder::{AstPass, QueryFragment};
+use types;
+
+pub trait IntoSingleTypeExpressionList<ST> {
+    type Expression;
+
+    fn into_single_type_expression_list(self) -> Self::Expression;
+}
+
+#[derive(Debug)]
```

Comment:
This should implement `Clone` and `Copy`.

Reply:
:+1: